instrumentation/servlet/servlet-3.0/javaagent-tests/src/test/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/jetty/JettyServletHandlerTest.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0.jetty;
+
+import static io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint.EXCEPTION;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.api.internal.HttpConstants;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpServerTestOptions;
+import io.opentelemetry.instrumentation.testing.junit.http.ServerEndpoint;
+import io.opentelemetry.javaagent.instrumentation.servlet.v3_0.AbstractServlet3Test;
+import io.opentelemetry.javaagent.instrumentation.servlet.v3_0.tomcat.TestServlet3;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.semconv.HttpAttributes;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.ErrorHandler;
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class JettyServletHandlerTest extends AbstractServlet3Test<Server, ServletHandler> {
+
+  @RegisterExtension
+  protected static final InstrumentationExtension testing =
+      HttpServerInstrumentationExtension.forAgent();
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    super.configure(options);
+    options.setHttpAttributes(
+        serverEndpoint -> {
+          Set<AttributeKey<?>> attributes =
+              new HashSet<>(HttpServerTestOptions.DEFAULT_HTTP_ATTRIBUTES);
+          attributes.remove(HttpAttributes.HTTP_ROUTE);
+          return attributes;
+        });
+    options.setContextPath("");
+    options.setTestNotFound(false);
+    options.setExpectedException(new ServletException(EXCEPTION.getBody()));
+  }
+
+  @Override
+  public boolean hasResponseSpan(ServerEndpoint endpoint) {
+    return (JettyServlet3Test.IS_BEFORE_94 && endpoint.equals(EXCEPTION))
+        || super.hasResponseSpan(endpoint);
+  }
+
+  @Override
+  protected SpanDataAssert assertResponseSpan(
+      SpanDataAssert span, SpanData parentSpan, String method, ServerEndpoint endpoint) {
+
+    if (JettyServlet3Test.IS_BEFORE_94 && endpoint.equals(EXCEPTION)) {
+      span.satisfies(it -> assertThat(it.getName()).matches(".*\\.sendError"))
+          .hasKind(SpanKind.INTERNAL);
```

Comment:
groovy code also asserts the parent span

Reply:
I had not expected this to be such a difficult PR...

here's why one of the parent assertions doesn't work:

old

```groovy
@Override
  void responseSpan(TraceAssert trace, int index, Object controllerSpan, Object handlerSpan, String method, ServerEndpoint endpoint) {
    if (IS_BEFORE_94 && endpoint == EXCEPTION) {
      sendErrorSpan(trace, index, handlerSpan)
    }
    super.responseSpan(trace, index, controllerSpan, handlerSpan, method, endpoint)
  }
```

new

```java
  @Override
  protected SpanDataAssert assertResponseSpan(
      SpanDataAssert span, SpanData parentSpan, String method, ServerEndpoint endpoint) {
    if (IS_BEFORE_94 && endpoint.equals(EXCEPTION)) {
      span.satisfies(it -> assertThat(it.getName()).matches(".*\\.sendError"))
          .hasKind(SpanKind.INTERNAL)
          .hasParent(parentSpan); // fails, because this corresponds to controllerSpan rather than handlerSpan
    }

    return super.assertResponseSpan(span, parentSpan, method, endpoint);
  }
```